### DISCUSSION TITLE
Declarative relation update

### DISCUSSION THREAD

**Discussion by sdn90** [prisma/prisma#3546] [Q&A] [answered]
I have a lot of schemas where I just want to declaratively list out the related rows. I don't want to deal with filtering out which ones to create or delete.

Is this currently possible? If not, do you guys think this feature is within the scope of Prisma?

This is how I'm imagining it to look:

```prisma
model BlogPost {
  id    Integer @id
  title String
  tags  Tag[]
}

model Tag {
  id         Integer   @id
  blogPostId Integer
  blogPost   BlogPost
  value      String
}
```

```typescript
// existing data
await prisma.blogPost.create({
  where: 1,
  data: {
    title: "hello",
    tags: {
      create: [
        { value: "Programming" },
        { value: "Node.js" },
        { value: "Javascript" }
      ]
    }
  }
})

// This should:
// Remove Tags "Node.js" and "Javacript"
// Create Tag "Web"
await prisma.blogPost.update({
  where: 1,
  data: {
    title: "hello",
    tags: {
      replace: [
        { value: "Programming" },
        { value: "Web"
      ]
   }
  }
})
```

**Comment by ryands17** [comment, marked as answer]
Hey @sdn90 👋 
Yes currently Prisma doesn't support this, so you would need to perform 2 queries to achieve this. 
It would be great if you could create a feature request for this [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
